### PR TITLE
frontend: navbar

### PR DESCRIPTION
**Pull request by LynxHack**
:tickets: **Ticket(s)**: Closes #
image

---

## :construction_worker: Changes

A brief summary of what changes were introduced.

## :flashlight: Testing Instructions

Explain how to test your changes, if applicable.

### CODE REVIEW

File: frontend/src/components/App.js [hunk 1]
```diff
@@ -2,42 +2,33 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { BrowserRouter, Route, Switch } from 'react-router-dom';
 import Pinpoint from 'pinpoint-client';
-import logo from '../assets/logo.svg';
+// import logo from '../assets/logo.svg';
```
Comment (bobheadxi) [inline frontend/src/components/App.js]: 😮 commented code?

File: frontend/src/components/Reset.js [hunk 2]
```diff
@@ -6,11 +6,11 @@ class Reset extends Component {
     this.state = {
       email: '',
     };
-    this.updatetextfields = this.updatetextfields.bind(this);
+    this.updateTextFields = this.updateTextFields.bind(this);
```
Comment (bobheadxi) [inline frontend/src/components/Reset.js]: 👍

File: frontend/src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+// import logo from './../assets'
```
Comment (bobheadxi) [inline frontend/src/components/Navbar.js]:
```suggestion
```

File: frontend/src/styles/utils/spacing.scss [hunk 4]
```diff
@@ -44,7 +47,9 @@ $spacing-xxl: 48px;
 
 .margin {
   // navbar
-  &-nav { margin-top: 1; }
+  &-nav {
```
Comment (bobheadxi) [inline frontend/src/styles/utils/spacing.scss]: refer to what i said verbally

**Review by bobheadxi — COMMENTED** [review]
niceeeeeeee

File: .gitattributes [hunk 5]
```diff
@@ -3,3 +3,8 @@
 
 # Force Unix EOLs, to prevent Windows from breaking them.
 *.* text eol=lf
+
+# Image Corruption Fix
+*.jpg binary
+*.png binary
```
Comment (bobheadxi) [inline .gitattributes]: 🚀

File: frontend/src/components/Navbar.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      <div className="pad-nav border-m">
+        <nav className="animate-menu navbar">
+          <div className="logo">
+            <a href="/">
+              <img src={logo} className="pinpointlogo" alt="logo" />
+            </a>
+          </div>
+          <ul className="margin-nav">
+            <li><a href="/">Home</a></li>
+            {this.checkLogin()}
```
Comment (bobheadxi) [inline frontend/src/components/Navbar.js]: `checkLogin` implies a condition, but this actually generates a div to display, maybe name something like `loginStateDiv`?

File: frontend/src/styles/utils/spacing.scss [hunk 7]
```diff
@@ -44,7 +47,9 @@ $spacing-xxl: 48px;
 
 .margin {
   // navbar
-  &-nav { margin-top: 1; }
+  &-nav { 
+    margin: 0 $spacing-s;
```
Comment (bobheadxi) [inline frontend/src/styles/utils/spacing.scss]: doesn't this set left-right spacing? I think `margin-nav` should only be used to set a space for page content *from the top of the page* - ie, avoid loading stuff covered by a nav bar

to set left-right and top-bottom, we should use dedicated margin classes, e.g. `margin-ends-none`

**Review by bobheadxi — COMMENTED** [review]
image

File: frontend/src/styles/utils/decorations.scss [hunk 8]
```diff
@@ -111,6 +110,32 @@ a {
   border-width: 3px;
 }
 
+.pinpointlogo {
+  float: left;
+  height: 60px;
+}
+
+// Navbar
```
Comment (leesw98) [inline frontend/src/styles/utils/decorations.scss]: Might want to move this to `styles/components/navbar.scss`

File: frontend/src/styles/utils/decorations.scss [hunk 9]
```diff
@@ -111,6 +110,32 @@ a {
   border-width: 3px;
 }
 
+.pinpointlogo {
```
Comment (leesw98) [inline frontend/src/styles/utils/decorations.scss]: Might want to split this into two separate files:
```
styles/components/positions.scss
styles/components/sizing.scss
```
and use the classes directly.

File: frontend/src/styles/utils/decorations.scss (hunk 8 above)
Comment (bobheadxi) [reply frontend/src/styles/utils/decorations.scss]: @LynxHack I second this

File: frontend/src/styles/utils/decorations.scss (hunk 9 above)
Comment (bobheadxi) [reply frontend/src/styles/utils/decorations.scss]: or even just `styles/components/navbar.scss` for now

**Review by bobheadxi — DISMISSED** [review]
minor things, but I agree with what @leesw98 brought up 👍 otherwise it looks almost good to go!

**Review by zhulia96 — DISMISSED** [review]

**Review by bwdmonkey — DISMISSED** [review]
refer to previous comment. (simply un-approving)

**Pull request merged**